examples_tests/common/CommonAPI.h

```diff
@@ -443,128 +681,256 @@ class CommonAPI
+		}
+
+		video::ISwapchain::SCreationParams sc_params = {};
 		sc_params.width = width;
 		sc_params.height = height;
 		sc_params.arrayLayers = 1u;
 		sc_params.minImageCount = imageCount;
-		sc_params.presentMode = presentMode;
+		sc_params.presentMode = requestedPresentMode;
+		sc_params.imageUsage = static_cast<asset::IImage::E_USAGE_FLAGS>(asset::IImage::EUF_COLOR_ATTACHMENT_BIT | asset::IImage::EUF_STORAGE_BIT | asset::IImage::EUF_TRANSFER_DST_BIT);;
 		sc_params.surface = surface;
-		sc_params.surfaceFormat.format = asset::EF_R8G8B8A8_SRGB;
-		sc_params.surfaceFormat.colorSpace.eotf = asset::EOTF_sRGB;
-		sc_params.surfaceFormat.colorSpace.primary = asset::ECP_SRGB;
+		sc_params.imageSharingMode = imageSharingMode;
+		sc_params.surfaceFormat = surfaceFormat;
 
 		return device->createSwapchain(std::move(sc_params));
 	}
-	static nbl::core::smart_refctd_ptr<nbl::video::IGPURenderpass> createRenderpass(const nbl::core::smart_refctd_ptr<nbl::video::ILogicalDevice>& device, nbl::asset::E_FORMAT depthFormat = nbl::asset::EF_UNKNOWN)
+	static nbl::core::smart_refctd_ptr<nbl::video::IGPURenderpass> createRenderpass(const nbl::core::smart_refctd_ptr<nbl::video::ILogicalDevice>& device, nbl::asset::E_FORMAT colorAttachmentFormat = nbl::asset::EF_UNKNOWN, nbl::asset::E_FORMAT depthFormat = nbl::asset::EF_UNKNOWN)
 	{
 		using namespace nbl;
 
 		bool useDepth = asset::isDepthOrStencilFormat(depthFormat);
 
 		video::IGPURenderpass::SCreationParams::SAttachmentDescription attachments[2];
 		attachments[0].initialLayout = asset::EIL_UNDEFINED;
-		attachments[0].finalLayout = asset::EIL_UNDEFINED;
-		attachments[0].format = asset::EF_R8G8B8A8_SRGB;
+		attachments[0].finalLayout = asset::EIL_PRESENT_SRC_KHR;
+		attachments[0].format = colorAttachmentFormat;
 		attachments[0].samples = asset::IImage::ESCF_1_BIT;
 		attachments[0].loadOp = video::IGPURenderpass::ELO_CLEAR;
 		attachments[0].storeOp = video::IGPURenderpass::ESO_STORE;
 
 		attachments[1].initialLayout = asset::EIL_UNDEFINED;
-		attachments[1].finalLayout = asset::EIL_UNDEFINED;
+		attachments[1].finalLayout = asset::EIL_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
 		attachments[1].format = depthFormat;
 		attachments[1].samples = asset::IImage::ESCF_1_BIT;
 		attachments[1].loadOp = video::IGPURenderpass::ELO_CLEAR;
 		attachments[1].storeOp = video::IGPURenderpass::ESO_STORE;
 
 		video::IGPURenderpass::SCreationParams::SSubpassDescription::SAttachmentRef colorAttRef;
 		colorAttRef.attachment = 0u;
-		colorAttRef.layout = asset::EIL_UNDEFINED;
+		colorAttRef.layout = asset::EIL_COLOR_ATTACHMENT_OPTIMAL;
 
 		video::IGPURenderpass::SCreationParams::SSubpassDescription::SAttachmentRef depthStencilAttRef;
 		depthStencilAttRef.attachment = 1u;
-		depthStencilAttRef.layout = asset::EIL_UNDEFINED;
+		depthStencilAttRef.layout = asset::EIL_DEPTH_STENCIL_ATTACHMENT_OPTIMAL;
 
 		video::IGPURenderpass::SCreationParams::SSubpassDescription sp;
+		sp.pipelineBindPoint = asset::EPBP_GRAPHICS;
```

Comment:
@Crisspl why does `subpassdescription` even have a pipeline bind point?
CC: @achalpandeyy is this something to do with mesh shaders?